core/src/execution/datafusion/spark_hash.rs

```diff
@@ -85,11 +85,16 @@ pub(crate) fn spark_compatible_murmur3_hash<T: AsRef<[u8]>>(data: T, seed: u32)
     // safety:
     // avoid boundary checking in performance critical codes.
     // all operations are guaranteed to be safe
+    // data is &[u8] so we do not need to check for proper alignment
     unsafe {
-        let mut h1 = hash_bytes_by_int(
-            std::slice::from_raw_parts(data.get_unchecked(0), len_aligned),
-            seed,
-        );
+        let mut h1 = if len_aligned > 0 {
```

Comment:
Without this check we were hitting a debug assertion on the call to `get_unchecked(0)`

Reply:
perhaps its time to rename the variable from h1 to something meaningful?